hi-ml-cpath/src/health_cpath/utils/output_utils.py

```diff
@@ -355,13 +355,13 @@ def save_validation_outputs(self, epoch_results: EpochResultsType, metrics_dict:
         :param epoch: Current epoch number.
         :param on_extra_val: Whether this is an extra validation epoch (e.g. after training).
```

Comment:
This is a very complicated logic, we should simplify this in the future. Creating an issue

Reply:
#766